src/lib/client.spec.ts

```diff
@@ -15,6 +15,264 @@
+    signature: 'blablabla',
+  };
+  it('should properly instantiate Client', () => {
+    const client = new Client(defaultApikey);
+    expect(client).toBeDefined();
+    expect(client).toBeInstanceOf(Client);
+  });
+  it('should throw error if api key not provided', () => {
+    expect(() => { return new Client(''); }).toThrow('An apikey is required to initialize the Filestack client');
+  });
+  it('should throw error if provided security without signature', () => {
+    const options = {
+      security: {
+        policy: 'examplePoolicy',
+        signature: '',
+      },
+    };
+    expect(() => { return new Client(defaultApikey, options); }).toThrow('Both policy and signature are required for client security');
+  });
+  it('should pass policy & signature to the session', () => {
+    const options = {
+      security: {
+        policy: 'examplePoolicy',
+        signature: 'blablabla',
+      },
+    };
+    const client = new Client(defaultApikey, options);
+    expect(client.session.policy).toBe(options.security.policy);
+    expect(client.session.signature).toBe(options.security.signature);
+  });
+  it('should pass cname to the session', () => {
+    const options = {
+      cname: 'testCname.com',
+    };
+    const client = new Client(defaultApikey, options);
+    expect(client.session.cname).toBe(options.cname);
+  });
+  it('should be able to logout for provied cloud', (done) => {
+    const client = new Client(defaultApikey);
+    const cloudName = 'dropboxx';
+    return client.logout(cloudName).then(() => {
+      expect.assertions(2);
+      expect(mockCloudLogout).toHaveBeenCalledTimes(1);
+      expect(mockCloudLogout).toHaveBeenCalledWith(cloudName);
+      done();
+    });
+  });
+  it('should get metadata for a provided handle', (done) => {
+    const client = new Client(defaultApikey);
+    const options = {
+      sha224: true,
+    };
+    const security = {
+      policy: 'examplePoolicy',
+      signature: 'blablabla',
+    };
+    return client.metadata(defaultHandle, options, security).then(() => {
+      expect.assertions(2);
+      expect(mockMetadata).toHaveBeenCalledTimes(1);
+      expect(mockMetadata).toHaveBeenCalledWith(client.session, defaultHandle, options, security);
+      done();
+    });
+  });
+  it('should be able to open picker', () => {
+    const client = new Client(defaultApikey);
+    const pickerOptions = {
+      lang: 'de',
+    };
+    client.picker(pickerOptions);
+    expect(mockPicker).toBeCalled();
+    expect(mockPicker).toBeCalledWith(expect.any(Object), pickerOptions);
+  });
+  it('should be able to open preview', () => {
+    const client = new Client(defaultApikey);
+    const previewOptions = {
+      id: 'testElement',
+    };
+    client.preview(defaultHandle, previewOptions);
+    expect(mockPreview).toBeCalled();
+    expect(mockPreview).toBeCalledWith(expect.any(Object), defaultHandle, previewOptions);
+  });
+  it('should be able to remove handle', (done) => {
```

Comment:
please add another test for remove (should throw an error during remove if security is not given)

Reply:
I only tested logic contained in this file and used mocks for external functions. 
This particular error throw which you mentioned is in ./api/file remove method.